CMakeLists.txt

```diff
@@ -626,6 +646,56 @@ if(CURL_FOUND)
 	endforeach()
 endif()
 
+else()#CI specifics
+
+#Platform specific hard links
```

Comment:
According to https://cmake.org/cmake/help/v3.15/_sources/command/file.rst.txt, it should be as easy as `file(CREATE_LINK <target> <link>`). No platform-specific code required, right?

Reply:
That only works in configuration phase, not the build phase. So you need to have git built before you can create the hardlinks. One way I can think of is create a separate cmake script file to create the hardlinks, which can be invoked after the git has been built, but this would still mess up the clean command. So I think doing platform specific stuff is better for now.

Reply:
What about using https://cmake.org/cmake/help/v3.16/manual/cmake.1.html#run-a-command-line-tool `create_symlink` to avoid OS specific code?

Reply:
When we create symlinks and tar the build artifacts and untar them for testing, tar is not able to resolve the symlinks. So that is why hardlinks were chosen.

Reply:
Hmm. What do other CMake-based projects do when they need to generate hard-links at build time?

Reply:
They don't as far as I know. Maybe I will try to add this feature to cmake.